src/System.Private.ServiceModel/tests/Scenarios/Contract/Data/DataContractTests.4.1.1.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using System.ServiceModel;
+using Infrastructure.Common;
+using Xunit;
+using System.Text;
+using System.ServiceModel.Description;
+using System.Runtime.Serialization;
+using System.Xml;
+using System.Linq;
+
+public static partial class DataContractTests
+{
+    [WcfFact]
+    [OuterLoop]
+    public static void DataContractResolverTest()
+    {
+        StringBuilder errorBuilder = new StringBuilder();
+        try
+        {
+            ChannelFactory<IWcfService> factory = new ChannelFactory<IWcfService>(new BasicHttpBinding(), new EndpointAddress(Endpoints.DataContractResolver_Address));
+            ContractDescription cd = factory.Endpoint.Contract;
+
+            foreach (var operation in factory.Endpoint.Contract.Operations)
+            {
+                DataContractSerializerOperationBehavior behavior =
+                         operation.OperationBehaviors.FirstOrDefault(
+                             x => x.GetType() == typeof(DataContractSerializerOperationBehavior)) as DataContractSerializerOperationBehavior;
+                behavior.DataContractResolver = new ManagerDataContractResolver();
+            }
+
+            IWcfService client = factory.CreateChannel();
+            client.AddEmployee(new Manager() { Age = "10", Name = "jone", OfficeId = 1 });
+            var results = client.GetAllEmployees();
+            Assert.Equal(1, results.Count());
+            var manager = results.First();
+            Assert.Equal(typeof(Manager), manager.GetType());
```

Comment:
Our preferred pattern is to use Assert.True(somecondition, some message) so that we can understand which Assert was triggered.  This is very good to help diagnose lab failures.